src/Tasks/Backup/BackupJob.php

```diff
@@ -2,276 +2,256 @@
+	}
 
-    protected function createZipContainingEveryFileInManifest(Manifest $manifest)
-    {
-        consoleOutput()->info("Zipping {$manifest->count()} files...");
+	protected function directoriesUsedByBackupJob(): array {
+		return $this->backupDestinations
+			->filter( function ( BackupDestination $backupDestination ) {
+				return $backupDestination->filesystemType() === 'local';
+			} )
+			->map( function ( BackupDestination $backupDestination ) {
+				return $backupDestination->disk()->getDriver()->getAdapter()->applyPathPrefix( '' ) . $backupDestination->backupName();
+			} )
+			->each( function ( string $backupDestinationDirectory ) {
+				$this->fileSelection->excludeFilesFrom( $backupDestinationDirectory );
+			} )
+			->push( $this->temporaryDirectory->path() )
+			->toArray();
+	}
 
-        $pathToZip = $this->temporaryDirectory->path(config('backup.backup.destination.filename_prefix').$this->filename);
+	protected function createZipContainingEveryFileInManifest( Manifest $manifest ) {
+		consoleOutput()->info( "Zipping {$manifest->count()} files..." );
 
-        $zip = Zip::createForManifest($manifest, $pathToZip);
+		$pathToZip = $this->temporaryDirectory->path( config( 'backup.backup.destination.filename_prefix' ) . $this->filename );
 
-        consoleOutput()->info("Created zip containing {$zip->count()} files. Size is {$zip->humanReadableSize()}");
+		$zip = Zip::createForManifest( $manifest, $pathToZip );
 
-        $this->sendNotification(new BackupZipWasCreated($pathToZip));
+		consoleOutput()->info( "Created zip containing {$zip->count()} files. Size is {$zip->humanReadableSize()}" );
 
-        return $pathToZip;
-    }
+		$this->sendNotification( new BackupZipWasCreated( $pathToZip ) );
 
-    /**
-     * Dumps the databases to the given directory.
-     * Returns an array with paths to the dump files.
-     *
-     * @return array
-     */
-    protected function dumpDatabases(): array
-    {
-        return $this->dbDumpers->map(function (DbDumper $dbDumper) {
-            consoleOutput()->info("Dumping database {$dbDumper->getDbName()}...");
+		return $pathToZip;
+	}
 
-            $dbType = mb_strtolower(basename(str_replace('\\', '/', get_class($dbDumper))));
+	/**
+	 * Dumps the databases to the given directory.
+	 * Returns an array with paths to the dump files.
+	 *
+	 * @return array
+	 */
+	protected function dumpDatabases(): array {
+		return $this->dbDumpers->map( function ( DbDumper $dbDumper ) {
+			consoleOutput()->info( "Dumping database {$dbDumper->getDbName()}..." );
 
-            $dbName = $dbDumper instanceof Sqlite ? 'database' : $dbDumper->getDbName();
+			$dbType = mb_strtolower( basename( str_replace( '\\', '/', get_class( $dbDumper ) ) ) );
 
-            $fileName = "{$dbType}-{$dbName}.sql";
+			$dbName   = $dbDumper instanceof Sqlite ? 'database' : $dbDumper->getDbName();
+			$fileName = $dbDumper instanceof MongoDb ? "{$dbType}-{$dbName}.archive" : "{$dbType}-{$dbName}.sql";
```

Comment:
Instead of having a special case for `MongoDb` introduce a new function `getExtension` that returns the extension to use.